src/crimson/osd/recovery_backend.cc

```diff
@@ -228,30 +228,27 @@ RecoveryBackend::scan_for_backfill(
   auto&& [objects, next] = co_await backend->list_objects(start, max);
   co_await interruptor::parallel_for_each(objects, [this, version_map]
     (const hobject_t& object) -> interruptible_future<> {
-    crimson::osd::ObjectContextRef obc;
-    if (pg.is_primary()) {
-      obc = pg.obc_registry.maybe_get_cached_obc(object);
-    }
-    if (obc) {
-      if (obc->obs.exists) {
-        logger().debug("scan_for_backfill found (primary): {}  {}",
-                       object, obc->obs.oi.version);
-        version_map->emplace(object, obc->obs.oi.version);
-      } else {
-        // if the object does not exist here, it must have been removed
-        // between the collection_list_partial and here.  This can happen
-        // for the first item in the range, which is usually last_backfill.
-      }
+    logger().debug("{} querying obj:{}", __func__, object);
+    auto obc_orderer = pg.obc_loader.get_obc_orderer(object);
+    auto obc_manager = pg.obc_loader.get_obc_manager(
```

Comment:
Shouldn't need an orderer -- https://github.com/ceph/ceph/pull/61561 introduces a constructor you should be able to use instead.